library/src/test/java/com/yayandroid/locationmanager/providers/locationprovider/DispatcherLocationProviderTest.java

```diff
@@ -295,6 +299,80 @@ public void resolveGooglePlayServicesShouldContinueWithDefaultWhenResolveDialogI
         verify(dispatcherLocationProvider).continueWithDefaultProviders();
     }
 
+    @Test
+    public void resolveGooglePlayServicesShouldContinueWithDefaultWhenErrorCannotBeResolved() {
+
+        int unresolvableError = ConnectionResult.SERVICE_INVALID;
+
+        final DialogInterface.OnDismissListener[] dismissListener = new DialogInterface.OnDismissListener[1];
+
+        when(dispatcherLocationSource.getGoogleApiErrorDialog(eq(activity), eq(unresolvableError),
+                eq(RequestCode.GOOGLE_PLAY_SERVICES), any(OnCancelListener.class))).thenReturn(dialog);
+
+        // catch and store real OnDismissListener listener
+        doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(InvocationOnMock invocation) {
+                dismissListener[0] = invocation.getArgument(0);
+
+                return null;
+            }
+        }).when(dialog).setOnDismissListener(any(DialogInterface.OnDismissListener.class));
+
+        // simulate dialog dismiss event
+        doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(InvocationOnMock invocation) {
+                dismissListener[0].onDismiss(dialog);
+
+                return null;
+            }
+        }).when(dialog).dismiss();
+
+        dispatcherLocationProvider.resolveGooglePlayServices(unresolvableError);
+
+        verify(dialog).show();
+
+        dialog.dismiss(); // Simulate dismiss dialog (error cannot be resolved)
+
+        verify(dispatcherLocationProvider).continueWithDefaultProviders();
+    }
+
+    @Test
+    public void resolveGooglePlayServicesShouldContinueWithDefaultWhenWhenResolveDialogIsCancelled() {
```

Comment:
Same as above

Reply:
To check that `continueWithDefaultProviders()` is called when dialog is cancelled.
`continueWithDefaultProviders()` called in OnCancelListener, so to simulate real listener call, I catch and store it. 
The `dialog` is `@Mock` object and `dialog.dismiss()` does nothing.